### PR TITLE
Migrate to studio

### PR DESCRIPTION
**Pull request by grooze**
Migration in Android Studio
Unit testes fixed
psUserIdsGC parameter was removed, Google AID was added
OpenUDID_manager.waitOpenUDIDInitialized rewritten

### CODE REVIEW

File: appmetr/src/main/java/com/appmetr/android/internal/LibraryPreferences.java [hunk 1]
```diff
@@ -43,19 +43,14 @@
...
-    /**
      * The maximum time in seconds to wait before shutdown ThreadPoolExecutor
      */
     public static final int THREAD_POOL_TERMINATION_TIMEOUT_IN_SEC = 5; // 5 seconds
 
     /**
      * The version string of AppMetrAndroid library
      */
-    public static final String VERSION_STRING = "1.8.4";
+    public static final String VERSION_STRING = "1.8.6";
```
Comment (chemist777) [inline appmetr/src/main/java/com/appmetr/android/internal/LibraryPreferences.java]: Why are version jumping to 1.8.6 instead of 1.8.5?

Comment (grooze) [inline appmetr/src/main/java/com/appmetr/android/internal/LibraryPreferences.java]: 1.8.5 already was in Unity repository at develop branch, but didn't commit to this repository

File: gradlew.bat [hunk 2]
```diff
@@ -1,90 +1,90 @@
...
+
+echo.
+echo ERROR: JAVA_HOME is not set and no 'java' command could be found in your PATH.
+echo.
+echo Please set the JAVA_HOME variable in your environment to match the
+echo location of your Java installation.
+
+goto fail
+
+:findJavaFromJavaHome
```
Comment (chemist777) [inline gradlew.bat]: All new line separators should be \n.

Comment (grooze) [inline gradlew.bat]: This is windows bat file! Why? It's autogenerated script

File: appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java [hunk 3]
```diff
@@ -301,7 +301,8 @@ protected void trackSessionImpl(JSONObject properties) {
                 properties = new JSONObject();
             }
 
-            long duration = mPreferences.getSessionDuration() / 1000;
+            long duration = (mPreferences.getSessionDuration() + (new Date().getTime() - mStartTime)) / 1000;
```
Comment (chemist777) [inline appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: Previously we tracked duration of previous session, not current session. But now we have got longer duration.

Comment (chemist777) [inline appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: I think you simply shouldn't send duration if it is zero.

Comment (grooze) [inline appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: What to do, if game starts new session without restarting application? Why session duration increased on pause / resume and new call trackSession after it?

File: gradlew.bat (hunk 2 above)
Comment (chemist777) [inline gradlew.bat]: Best practice is to store only \n as line endings in git.

File: appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java (hunk 3 above)
Comment (chemist777) [inline appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: We should monitor the full duration of the session. We don't know duration at first launch (trackSession) of application. So we should send previous session duration on subsequent launches (trackSession) or don't send duration attribute in case of first run.

Comment (grooze) [inline appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: I say nothing about first launch. If application can track only one session at launch, it must be called automatically without public api

Comment (chemist777) [inline appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: Library users want to pass additional attributes to trackSession method. So we must leave it public.

Comment (grooze) [inline appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: Why this call on application pause:
`mPreferences.setSessionDuration(mPreferences.getSessionDuration() + (new Date().getTime() - mStartTime));`?
If game call trackSession after application pause, AppMetr starts new session. If game call trackSession without application pause, nothing happend, only faked zero-length session request. Current session duration will flows without changes.

Comment (chemist777) [inline appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: This strange behavior should be changed. I don't know how to fix it yet.

Comment (chemist777) [inline appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: I think you should remove these changes (session duration) and make another pull request with new session duration logic.

File: gradlew.bat (hunk 2 above)
Comment (grooze) [inline gradlew.bat]: Fixed

File: appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java (hunk 3 above)
Comment (grooze) [inline appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java]: Ok, removed
